### PR TITLE
feat: customizable lock screen & hard block + Built-in Kotlin migration

### PR DESCRIPTION
**Pull request by thorito**
Thanks for this excellent library, it has been really useful for protecting sensitive content in our Flutter apps.

This PR adds two features:

## 1. Customizable lock screen & hard block UI

Two new optional builders added to SecureContentScope, using the same pattern as the existing overlayBuilder:

```dart
SecureContentScope(
  enabled: true,
  lockScreenBuilder: (context, onUnlock) => MyCustomLockScreen(
    onUnlock: onUnlock,
  ),
  hardBlockBuilder: (context) => MyCustomHardBlock(),
  child: MyApp(),
)
```

- lockScreenBuilder — receives onUnlock callback that handles both regular unlock and biometric re-auth
- hardBlockBuilder — replaces the hard-block screen (integrity risk)

When not provided, the current built-in UI is used as fallback — zero breaking changes.

## 2. Built-in Kotlin migration

Migrated the Android plugin and example app to Built-in Kotlin following Flutter's migration guide:

- Removed explicit KGP application from plugin
- Replaced kotlinOptions with kotlin { compilerOptions { ... } }
- Removed KGP declarations from example app
- Eliminates the KGP build warning

## Files changed (13)

- lib/src/secure_content_scope.dart — LockScreenBuilder / HardBlockBuilder typedefs + builders
- lib/secure_content.dart — Export new typedefs
- example/lib/main.dart — Demo of custom lock screen & hard block
- android/build.gradle — Built-in Kotlin migration
- android/settings.gradle — Add pluginManagement
- example/android/settings.gradle — Remove explicit KGP
- example/android/app/build.gradle — Remove kotlin-android plugin
- pubspec.yaml — Bump to 2.0.0-beta.3
- CHANGELOG.md — Changelog for beta.3

Let me know if you'd like any adjustments!

### CODE REVIEW

**Review by codenameakshay — CHANGES_REQUESTED** [review]
Thanks for the PR, @thorito — the lock-screen/hard-block builder API is well-designed: it mirrors the existing `overlayBuilder` pattern, the `_handleUnlock` extraction cleanly de-duplicates the biometric-vs-plain branch, and the null-fallback keeps it non-breaking. 🙏

Before I can merge, a few things need to be addressed. I've split them into **required** vs. **recommended**.

## Required for merge

**1. Verify (and likely fix) the Built-in Kotlin migration — it looks internally contradictory.**
`example/android/gradle.properties` sets `android.builtInKotlin=false` and `android.newDsl=false`, *while* the explicit `org.jetbrains.kotlin.android` plugin is removed from both `example/android/settings.gradle` and `example/android/app/build.gradle`. But:
- the example still has a Kotlin `MainActivity.kt`, and
- the plugin's `android/build.gradle` now uses a `kotlin { compilerOptions { ... } }` block that requires the Kotlin Gradle plugin to be applied to the module.

With built-in Kotlin disabled *and* the explicit plugin removed, there's no obvious source for the Kotlin plugin (the example build also compiles the plugin module as a path dependency), so both the example's Kotlin and the plugin's `kotlin {}` extension would normally fail to resolve. If the intent is to adopt built-in Kotlin, these flags should be `true`. Please run `cd example && flutter build apk` and confirm it builds cleanly, and adjust the flags accordingly.

**2. Guarantee the custom screens actually obscure protected content (security).**
The default lock/hard-block screens use an opaque `ColoredBox`. A custom `hardBlockBuilder`/`lockScreenBuilder` that is translucent or doesn't fill the area would let the sensitive `child` show through underneath — which defeats the integrity hard-block. Please:
- document that these builders **must be fully opaque and fill the available space**, and
- wrap both overlays in `AbsorbPointer` (or an opaque gesture barrier) so taps in the margins around the builder's content don't fall through the `Stack` to the protected `child`. (This pointer-passthrough is pre-existing, but this PR is the right place to harden it since it touches exactly this code.)

**3. Add test coverage for the new API.**
There's currently no `test/` for the package. Please add at least a widget test covering: (a) default UI renders when the builders are null, (b) the custom builder renders when provided, and (c) `onUnlock` actually unlocks. For a security-focused library this is important.

## Recommended

- **CHANGELOG**: the `beta.3` entry only mentions the two builders — please also note the Kotlin migration and the dependency/gradle bumps shipping in the same version.
- **Pigeon bump (26 → 27, a major)** without regenerating the committed `*.g.dart`/`*.g.kt` output is inconsistent. Either regenerate or leave `pigeon` pinned. Also please confirm `flutter analyze` is clean under `flutter_lints: ^6.0.0`.
- **Consider splitting** the Android build migration into its own PR so the (merge-ready) API addition isn't blocked on the Gradle verification.

## Nits

- Add dartdoc to the new public symbols (`LockScreenBuilder`, `HardBlockBuilder`, and the two new fields) — good for the pub.dev API score and a natural place to state the opacity requirement from #2.
- `_buildCustomLockScreen` in `example/lib/main.dart` has slightly off indentation vs. `_buildCustomHardBlockScreen`.

Happy to re-review once #1–#3 are addressed. Thanks again for contributing!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 2.0.0-beta.3
+
+- Added `lockScreenBuilder` and `hardBlockBuilder` to `SecureContentScope` for custom lock screen and hard-block UI.
+- Migrated example Android project to built-in Kotlin (AGP 8.12 / Gradle 8.13 / Java 21).
```
Comment (codenameakshay) [inline CHANGELOG.md:4]: The example Gradle wrapper is bumped to **8.14** in this PR (`gradle-wrapper.properties`), but this line says 8.13.

```suggestion
- Migrated example Android project to built-in Kotlin (AGP 8.12 / Gradle 8.14 / Java 21).
```

**Review by codenameakshay — APPROVED** [review]
Re-reviewed the updated commit — all three required items are resolved, thanks for the thorough turnaround. 🎉

- **Kotlin migration**: `builtInKotlin`/`newDsl` are now both `true`, consistent with the migration intent.
- **Security hardening**: both overlays wrapped in `GestureDetector(behavior: HitTestBehavior.opaque)` + opacity warnings in the dartdoc. Nice call using an opaque `GestureDetector` rather than `AbsorbPointer` — it absorbs the margin taps without disabling the unlock button inside the lock screen.
- **Tests**: `test/secure_content_scope_test.dart` covers default UI, both custom builders, and both unlock paths.
- **Pigeon** genuinely regenerated to v27.1.0 (Dart/Kotlin/Swift headers all updated), CHANGELOG + dartdoc + example indentation all done.
- I also see you reverted the `StreamController.broadcast(sync: true)` change from the earlier push — 👍 that was the one I was going to ask you to drop (it introduced synchronous reentrancy via `_unlock() → emitLocalEvent`). Reverting to async is the right call.

**Approving.** Two small things to wrap up before you merge (non-blocking):

1. CHANGELOG version typo — see the inline suggestion (Gradle was bumped to 8.14, not 8.13).
2. This repo has **no CI**, so the Android build is still unverified. Please run `cd example && flutter build apk` once locally to confirm the built-in-Kotlin flags compile cleanly before merging.

Tiny nit, no action needed: the added `break;` statements are harmless, but there wasn't actually a fall-through bug — Dart 3 `switch` statements don't fall through, so the original already behaved correctly.

Thanks again for the contribution and the clean follow-up!
